frontend_tests/node_tests/compose.js

```diff
@@ -554,6 +555,121 @@ people.add(bob);
     assert(compose_finished_event_checked);
 }());
 
+(function test_initialize() {
+    // In this test we mostly do the setup stuff in addition to testing the
+    // normal workflow of the function. All the tests for the on functions are
+    // done in subsequent tests directly below this test.
+
+    var resize_watch_manual_resize_checked = false;
+    resize.watch_manual_resize = function (elem) {
+        assert.equal('#new_message_content', elem);
+        resize_watch_manual_resize_checked = true;
+    };
+    global.window = {
+        XMLHttpRequest: true,
+        bridge: true,
+    };
+    var xmlhttprequest_checked = false;
+    set_global('XMLHttpRequest', function () {
+        this.upload = true;
+        xmlhttprequest_checked = true;
+    });
+    $("#compose #attach_files").addClass("notdisplayed");
+    feature_flags.dropbox_integration = true;
+    var dropbox_integration_loaded = false;
+    global.LazyLoad.js = function (script, on_success) {
+        assert.equal(script, 'https://www.dropbox.com/static/api/1/dropins.js');
+        dropbox_integration_loaded = true;
+
+        blueslip.error = function (error_msg) {
+            assert.equal(error_msg, 'Dropbox script reports loading but window.Dropbox undefined');
+        };
+        on_success();
+
+        global.window.Dropbox = 'fake-dropbox-script';
+        global.window.dropboxAppKey = 'fake-dropbox-appkey';
+        var dropbox_isbrowsersupported_checked = false;
+        global.Dropbox = {
+            isBrowserSupported: function () {
+              dropbox_isbrowsersupported_checked = true;
+              return true;
+            },
+        };
+        var dropbox_init_checked = false;
+        global.Dropbox.init = function (data) {
+            assert.equal(data.appKey, 'fake-dropbox-appkey');
+            dropbox_init_checked = true;
+        };
+        $("#compose #attach_dropbox_files").addClass("notdisplayed");
+        on_success();
+        assert(dropbox_isbrowsersupported_checked);
+        assert(dropbox_init_checked);
+        assert(!$("#compose #attach_dropbox_files").hasClass("notdisplayed"));
+    };
+    global.document = 'fake-document-object';
+    global.csrf_token = 'fake-csrf-token';
+    var filedrop_in_compose_checked = false;
+    page_params.maxfilesize = 512;
+    $("#compose").filedrop = function (payload) {
```

Comment:
You can extract a method called `verify_filedrop_parms here` to help de-clutter this test a bit.  You'll still need a small function here to set the var `filedrop_in_compose_checked` to true, but let it delegate most of the details to the helper function.